src/run_assemble.py

```diff
@@ -18,12 +20,55 @@
 from paths.assemble_output_dir import AssembleOutputDir
 from system import console
 
+def generate_manifest_file(args: AssembleArgs):
+    component = args.component or "opensearch"
+    platform_name = args.platform or platform.system().lower()
+    arch = args.arch or platform.machine().lower()
+    version = args.version or "0.0.1"
+    dist = args.dist or "tar"
+
+    url = f"https://artifacts.opensearch.org/releases/bundle/{component}/{version}/{component}-{version}-{platform_name}-{arch}.{dist}"
```

Comment:
We should not hard code this to prod url as if you see other similar features, we only use prod url substitutions in Jenkins lib. As opensearch-build was meant to run in any env and expected the files to be linked to local file paths.